indy_node/utils/node_control_utils.py

```diff
@@ -82,22 +226,93 @@ def _pkts_dedup(cls, deps):
         return ret
 
     @classmethod
-    def _parse_version_deps_from_pkt_mgr_output(cls, output):
+    def _parse_version_deps_from_pkg_mgr_output(
+            cls,
+            output: str,
+            upstream_cls: Type[Union[SourceVersion, None]] = None
+    ):
         out_lines = output.split("\n")
         ver = None
         ext_deps = []
+        num_pkgs = 0
         for ln in out_lines:
             act_line = ln.strip(" \n")
             if act_line.startswith("Version:"):
-                ver = ver or act_line.split(":", maxsplit=1)[1].strip(" \n")
+                # this method might be used for the dependnecy tree resolving
+                # when 'output' includes data for multiple packages,
+                # version info here doesn't make any sense in such a case
+                num_pkgs += 1
+                ver = (None if num_pkgs > 1 else
+                       act_line.split(":", maxsplit=1)[1].strip(" \n"))
             if act_line.startswith("Depends:"):
                 ext_deps += cls._parse_deps(act_line.split(":", maxsplit=1)[1].strip(" \n"))
-        return ver, cls._pkts_dedup(ext_deps)
+
+        if ver and upstream_cls:
+            try:
+                ver = DebianVersion(ver, upstream_cls=upstream_cls)
+            except InvalidVersionError as exc:
+                logger.warning(
+                    "Failed to parse debian version {}: {}"
+                    .format(ver, exc)
+                )
+                ver = None
+                ext_deps = []
+        else:
+            ver = None
+
+        return ver, cls._pkgs_dedup(ext_deps)
 
     @classmethod
-    def curr_pkt_info(cls, pkg_name):
+    def curr_pkg_info(cls, pkg_name: str) -> Tuple[PackageVersion, List]:
         package_info = cls._get_curr_info(pkg_name)
-        return cls._parse_version_deps_from_pkt_mgr_output(package_info)
+        return cls._parse_version_deps_from_pkg_mgr_output(
+            package_info, upstream_cls=src_version_cls(pkg_name))
+
+    @classmethod
+    def get_latest_pkg_version(
+            cls,
+            pkg_name: str,
+            upstream: SourceVersion = None,
```

Comment:
Do we have tests where upstream is specified?

Reply:
`indy_node/test/node_control_utils/test_node_control_util.py::test_generated_cmd_get_latest_pkg_version` it checks generated shell command when `upstream` is specified, further logic doesn't depend on `upstream`